### PR TITLE
Display duration on Preview and Detail View for a segment of a TV/radio transcript

### PR DESCRIPTION
**Pull request by dzonidoo**
CPCN-31

## Checklist

- [x] This pull request is not adding new forms that use redux
- [x] This pull request is adding missing TypeScript types to modified code segments where it's easy to do so with confidence
- [x] This pull request is replacing `lodash.get` with optional chaining for modified code segments

### CODE REVIEW

File: assets/wire/components/fields/DurationLabel.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    
+        return value;
+    }
+
+    const label = get(filterGroupLabels, 'duration', gettext('Duration'));
+    const time = secondsToHHMMSS(item.extra.duration);
+    const text = label + ': ' + time;
+
+    return (
+        <span>{text}</span>
```
Comment (petrjasek) [inline assets/wire/components/fields/DurationLabel.tsx:36]: we have some classes on the other fields, shouldn't we use those here as well?

File: assets/wire/components/fields/DurationLabel.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        const h = zeroPad(Math.floor(seconds / 3600));
+        const m = zeroPad(Math.floor(seconds % 3600 / 60));
+        const s = zeroPad(Math.floor(seconds % 3600 % 60));
+
+        const value = `${h}:${m}:${s}`;
+    
+        return value;
+    }
+
+    const label = get(filterGroupLabels, 'duration', gettext('Duration'));
```
[the remark this answers is not in the document]
Comment (dzonidoo) [reply assets/wire/components/fields/DurationLabel.tsx:31]: so label is just `const label = gettext('Duration');`

[the remark this answers is not in the document]
Comment (petrjasek) [reply assets/wire/components/fields/DurationLabel.tsx:31]: you could use `filterGroupLabels?.duration ?? : gettext('Duration');`

File: assets/wire/components/fields/DurationLabel.tsx (hunk 1 above)
Comment (dzonidoo) [reply assets/wire/components/fields/DurationLabel.tsx:36]: we just use classname for UrgencyLabel in some specific case

File: assets/wire/components/fields/DurationLabel.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {gettext} from 'utils';
+
+export function DurationLabel ({item, filterGroupLabels}: any) {
```
Comment (MarkLark86) [inline assets/wire/components/fields/DurationLabel.tsx:5]: @dzonidoo Please add types as per the PR checklist. This includes adding `duration` to the `IArticle['extra']` (https://github.com/superdesk/newsroom-core/blob/develop/assets/interfaces/content.ts#L26), and avoiding use of `any` and `PropType` and instead use the appropriate `IProps` interface to be used for this component

File: assets/wire/components/fields/DurationLabel.tsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+export function DurationLabel (props: IProps) {
+    const {item, filterGroupLabels} = props;
+
+    if (item.extra?.duration == null) {
+        return null;
+    }
+
+    function zeroPad(number: number) {
```
Comment (petrjasek) [inline assets/wire/components/fields/DurationLabel.tsx:20]: imo would be better to make it return string :
`zerePad(number: number) : string {`

**Review by petrjasek — APPROVED** [review]

**Review by petrjasek — APPROVED** [review]

**Pull request merged**